craft.lic

```diff
@@ -0,0 +1,291 @@
+    elsif rank <= 175 # Tier 4 - Simple
+      smith('a metal ankle band')
+    elsif rank <= 300 # Tier 5 - Basic
+      smith('a metal lockpick ring')
+    elsif rank <= 425 # Tier 6 - Somewhat Challenging
+      smith('a metal armband')
+    elsif rank <= 550 # Tier 7 - Challenging
+      smith('some metal clippers')
+    elsif rank <= 700 # Tier 8 - Complicated
+      smith('some squat knitting needles')
+      # Journeyman book: triangular wire sieve, bent <metal> scissors, knobby sewing needles, squat knitting needles, narrow <metal> awl
+    elsif rank <= 850 # Tier 9 - Intricate
+      smith('serrated hide scraper')
+      # Master book: beveled wood shaper, serrated hide scraper, compact <metal> awl, compact <metal> awl, round pestle
+    elsif rank <= 1175 # Tier 10 - Difficult
+      smith('serrated scissors')
+      # Master book: slender <metal> awl, serrated scissors, grooved pestle, oblong wire sieve
+    elsif rank <= 1400 # Tier 11 - Very Difficult
+      smith('thin sewing needles')
+      # Master book: jagged wood shaper, thin sewing needles
+    else # Tier 12 - Extremely Difficult
+      smith('trapezoidal wire sieve')
+      # Master book: trapezoidal wire sieve
+    end
+  end
+
+  def train_alchemy
+    if @settings.train_workorders.include?('Remedies')
+      return unless money_for_training?(5000, 'Alchemy')
+      wait_for_script_to_complete('workorders', ['Remedies'])
+      wait_for_script_to_complete('sell-loot')
+      walk_to(@training_room)
+      return
+    end
+
+    beep
+    echo('*** NOT YET IMPLEMENTED ***')
+    @settings.crossing_training.delete('Alchemy')
+  end
+
+  def money_for_training?(amount, skill)
+    if ensure_copper_on_hand(amount, @settings)
+      true
+    else
+      echo("Low on funds, removing #{skill} from training")
+      @settings.crossing_training.delete(skill)
+      false
+    end
+  end
+
+  def sew(chapter, unique_name, item)
+    check_yarn
+
+    walk_to(@training_room)
+    check_listening
+    check_teaching
+
+    wait_for_script_to_complete('sew', ['trash', 'knitting', chapter, unique_name, item, 'wool'])
+
+    bput('stow my yarn', 'You put', 'Stow what')
+  end
+
+  def shape(chapter, unique_name, item)
+    check_wood
+
+    find_shaping_room(@hometown, @training_room)
+    check_listening
+    check_teaching
+
+    wait_for_script_to_complete('shape', ['trash', chapter, unique_name, 'maple', item])
+
+    bput('get my lumber', 'You get some', 'You pick up')
+    dispose_trash('lumber')
+  end
+
+  def smith(full_name)
+    wait_for_script_to_complete('smith', ['bronze', full_name, 'buy'])
+    dispose_trash(right_hand)
+    dispose_trash(left_hand)
+  end
+
+  def buy_wood
+    crafting_data = get_data('crafting')
+    return unless money_for_training?(1000, 'Engineering')
+    order_item(crafting_data['shaping'][@hometown]['stock-room'], 10)
+    bput('stow my lumber', 'You put your', 'Stow what')
+  end
+
+  def buy_yarn(skipcoin = false)
```

Comment:
❓ I don't see any code paths in which `buy_yarn` is called with an argument. Why is `skipcoin` here?

Reply:
line 231, so if you only have 3 gold you dont run to the bank after each purchase

Reply:
Sure, line 231 (or, rather, now line 190). But that passes along `skipcoin` to itself recursively. To clarify my question, I don't see anywhere that `skipcoin` is changed from its default of `false`. Should line 190 be `true`?

Reply:
so I wasn't sure about this, and from testing I saw passing anything but false there made it true and skip withdrawing. buy_yarn(true) does work too, i'll change it to that

Reply:
fixed